lib/opensusebasetest.pm

```diff
@@ -318,6 +318,86 @@ sub handle_uefi_boot_disk_workaround {
     wait_screen_change { send_key 'ret' };
 }
 
+=head2 wait_grub
+
+  wait_grub([bootloader_time => $bootloader_time] [,in_grub => $in_grub]);
+
+Makes sure the bootloader appears. Returns successfully when reached the bootloader menu, ready to control it further or continue. The time waiting for the bootloader can be configured with
+C<$bootloader_time> in seconds. Set C<$in_grub> to 1 when the
+SUT is already expected to be within the grub menu.
+=cut
+sub wait_grub {
+    my ($self, %args) = @_;
+    my $bootloader_time = $args{bootloader_time} // 100;
+    my $in_grub         = $args{in_grub}         // 0;
+    my @tags            = ('grub2');
+    push @tags, 'bootloader-shim-import-prompt'   if get_var('UEFI');
+    push @tags, 'boot-live-' . get_var('DESKTOP') if get_var('LIVETEST');             # LIVETEST won't to do installation and no grub2 menu show up
```

Comment:
this is really messy , wait_grub has option to not wait for grub but do something else ? :(

Reply:
So what do you want? I extract `wait_grub` because this is what it does. AFAIU in case of live media it's still "grub" but with different needle tags.